### PR TITLE
Added package BuildTools.FxCop

### PR DESCRIPTION
**Pull request by HenrikRxn**
Part of #4 

@parthopdas 
I have some questions: 
1) Should we keep checking packages in  or should we switch to using package restore?
2) I have been doing some reading and it appears that FxCop operates on IL level, i.e. could be useful for F#, but I have not found any F# ruleset yet.
3) Is there a reason for using 'Visual Studio 2013' as Appveyor image and not 2015?

There is also the issue of switching to newer (though still in development) FxCop packages from the Roslyn team, e.g. https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers/

### CODE REVIEW

**Comment by HenrikRxn** [comment]
Ready for review. Please squash as the history is not interesting, mostly trial and error because of differences in my local environment (e.g. VS 2015 and C# Essentials extensions) and the Appveyor image (VS2013)

File: Common/Domain/Domain.fsproj [hunk 1]
```diff
@@ -24,15 +25,21 @@
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <WarningLevel>3</WarningLevel>
     <OtherFlags>--keyfile:..\..\tddstud10.snk</OtherFlags>
+    <RunCodeAnalysis>False</RunCodeAnalysis>
```
Comment (parthopdas) [inline Common/Domain/Domain.fsproj]: should this not be true? same with others

File: Hosts/Console/TddStud10App/TddStud10App.csproj [hunk 2]
```diff
@@ -180,6 +189,7 @@
       <Generator>ResXFileCodeGenerator</Generator>
       <LastGenOutput>Resources.Designer.cs</LastGenOutput>
     </EmbeddedResource>
+    <None Include="TddStud10AppCodeAnalysisRules.ruleset" />
```
Comment (parthopdas) [inline Hosts/Console/TddStud10App/TddStud10App.csproj]: shouldnt we merge all the rulesets into 1 file at the top level and use that for all the projects?

File: Common/Domain/Domain.fsproj (hunk 1 above)
Comment (parthopdas) [inline Common/Domain/Domain.fsproj]: i get it now, for f# they have no effect (fsharplint is for f#). hence false - in that case why not just revert the changes for f# projects?

**Comment by parthopdas** [comment]
w.r.t. checking in the packages, I don't think we should. I think we should
switch to paket (built on nuget, but with stricter dependency management) -
it is one todo-s i mentioned in #4. It will be great if you can pick that
up as well, in addition to the fxcop and fsharplint.

File: Common/Domain/Domain.fsproj (hunk 1 above)
Comment (HenrikRxn) [inline Common/Domain/Domain.fsproj]: I'll revert changes to the F# project files as you suggested. I naively assumed that uninstalling the FxCop NuGet reference would get things back to the way they were before

**Comment by parthopdas** [comment]
I am having trouble logging into gh from sydney. Not sure what the issue is. Hence responding to your queries in bits and pieces.

@parthopdas 
I have some questions: 
> 1) Should we keep checking packages in or should we switch to using package restore?
>>No in my opinion, we should switch to paket and delete the packages

> 2) I have been doing some reading and it appears that FxCop operates on IL level, i.e. could be useful for F#, but I have not found any F# ruleset yet.
>>FxCop get confused with IL generated by F#. For F# we have FSharpLint (analogous to a combo of StyleCop + FxCop for C#). It is a part of #4, please look into integrating it as well. 

> 3) Is there a reason for using 'Visual Studio 2013' as Appveyor image and not 2015?
>>its complicated :P some TddStud10Package takes dependency on some of the MPF v12 assemblies. But those are versioned as v14 on when building on VS2015 machines. And then that vsix cannot be loaded in vs2013 as vs2013 has only v12 of the MPF assemblies. I feel we should continue supporting VS2013, though we can mandate the builds to be done on VS2015 - do you see any benifits in buildin on vs2015 machines?

File: Hosts/Console/TddStud10App/TddStud10App.csproj (hunk 2 above)
Comment (HenrikRxn) [inline Hosts/Console/TddStud10App/TddStud10App.csproj]: There is nothing preventing us from doing this. I just didn't want to make that decision, but will put it in the solution files?

**Comment by HenrikRxn** [comment]
Will address StyleCop, FSharpLint, etc. in separate PRs, but want to keep it small and easier to revert should the need arise

File: Hosts/Console/TddStud10App/TddStud10App.csproj (hunk 2 above)
Comment (parthopdas) [inline Hosts/Console/TddStud10App/TddStud10App.csproj]: Yep at sln level a common ruleset for all C# projects.

**Comment by parthopdas** [comment]
Understood and agreed. Keeping PRs small is a good idea.

**Comment by parthopdas** [comment]
Once you have unified the rulesets and reverted the F# changes, please feel free to merge the pull request into master.

**Comment by HenrikRxn** [comment]
@parthopdas Ready to merge PR, but one last thing: Would you prefer to add the shared FxCopCSharp.ruleset as a link reference in each of the C# projects? As it is now you have to go to the "Code Analysis" section of the project properties to find out which ruleset is used for analysis

**Comment by parthopdas** [comment]
By "link reference" do you mean like the way it is currently - i.e. each projects refers the ruleset by a relative path? Or did you mean each project has its own ruleset and they all link to the top level ruleset?

IMO the way it is currently is good enough as I don't see any downsides.

So please go ahead merge as soon as you are ready.

**Comment by parthopdas** [comment]
merged the pr as i had a bunch of commits on my box that i wanted to rebase on top of your changes.
